MonoGame.Framework/Graphics/SpriteBatcher.cs

```diff
@@ -292,8 +292,14 @@ private void FlushVertexArray(int start, int end, Effect effect)
                 var passes = effect.CurrentTechnique.Passes;
                 foreach (var pass in passes)
                 {
+                    // Whatever happens in pass.Apply, make sure the texture being drawn
+                    // ends up in Textures[0].
+                    var texture = _device.Textures[0];

```

Comment:
Can you instead pass the current texture thru to `FlushVertexArray` so you have it here to begin with?